File: spring-java/src/main/java/com/jmpark/springjava/entity/PayMethod.java
package com.jmpark.springjava.entity;

import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.RequiredArgsConstructor;
import lombok.ToString;

import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.IdClass;

@Entity
@RequiredArgsConstructor
@Getter
@ToString
@EqualsAndHashCode(callSuper = false)
@IdClass(AreaId.class)
public class PayMethod extends BaseTimeEntity {
    @Id
    Integer areaId = 0;
    @Id
    String type = "";
}

File: spring-kotlin/src/main/kotlin/com/jmpark/springkotlin/entity/PayMethod.kt
package com.jmpark.springkotlin.entity

import javax.persistence.Entity
import javax.persistence.Id
import javax.persistence.IdClass

@Entity
@IdClass(AreaId::class)
data class PayMethod (
    @Id
    val areaId: Int = 0,
    @Id
    var type: String = "",
) : BaseTimeEntity()